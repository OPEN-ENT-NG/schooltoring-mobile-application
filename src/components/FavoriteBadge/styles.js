import { StyleSheet, Platform } from "react-native";

import { COLORS } from "../../styles/common";

const shadow =
  Platform.OS === "android"
    ? {
        elevation: 4
      }
    : {
        shadowColor: COLORS.GREY,
        shadowOffset: { width: 2, height: 2 },
        shadowOpacity: 0.6,
        shadowRadius: 1
      };

export default StyleSheet.create({
  badge: {
    alignItems: "stretch",
    width: "50%",
    padding: 10
  },
  top: {
    aspectRatio: 1,
    justifyContent: "center",
    alignItems: "center",
    backgroundColor: COLORS.SECONDARY_BACKGROUND,
    ...shadow
  },
  bottom: {
    backgroundColor: COLORS.BACKGROUND,
    ...shadow,
    maxHeight: 120
  },
  name: {
    textAlign: "center",
    color: COLORS.TEXT,
    fontWeight: "bold",
    fontSize: 16
  },
  buttons: {
    flexDirection: "row",
    justifyContent: "space-around",
    alignItems: "center",
    padding: 5
  },
  chatButton: {
    width: 40,
    height: 40,
    borderRadius: 20,
    backgroundColor: COLORS.PRIMARY
  },
  loader: {
    ...StyleSheet.absoluteFillObject,
    backgroundColor: "rgba(255, 255, 255, 0.8)",
    elevation: 1000,
    zIndex: 5
  }
});
