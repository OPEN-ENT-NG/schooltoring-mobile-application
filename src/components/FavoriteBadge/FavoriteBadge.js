import React from "react";
import { View, Text } from "react-native";
import PropTypes from "prop-types";

import Icon from "react-native-vector-icons/MaterialIcons";

import styles from "./styles";

import Avatar from "../Avatar/Avatar";
import SecondaryButton from "../SecondaryButton/SecondaryButton";
import Loader from "../Loader/Loader";

const FavoriteBadge = props => {
  return (
    <View style={styles.badge}>
      <View style={styles.top}>
        <Avatar
          size={140}
          src={`${global.config.auth.endpoint}${props.userinfo.avatar}`}
        />
      </View>
      <View style={styles.bottom}>
        <View style={styles.buttons}>
          <SecondaryButton id="on-delete-button" onPress={props.onDelete}>
            <Icon size={32} color={"red"} name="favorite" />
          </SecondaryButton>
          <SecondaryButton
            id="on-chat-button"
            style={styles.chatButton}
            onPress={props.onChat}
            disabled={!!props.chatDisabled}
          >
            <Icon size={20} color={"white"} name={"chat"} />
          </SecondaryButton>
        </View>
        <View style={{ paddingHorizontal: 10, paddingBottom: 15 }}>
          <Text style={styles.name} id="username-field">
            {props.userinfo.username}
          </Text>
        </View>
      </View>
      {props.loading && <Loader style={styles.loader} />}
    </View>
  );
};

FavoriteBadge.propTypes = {
  userinfo: PropTypes.shape({
    username: PropTypes.string.isRequired,
    avatar: PropTypes.string
  }).isRequired,
  loading: PropTypes.bool.isRequired,
  onDelete: PropTypes.func.isRequired,
  onChat: PropTypes.func.isRequired,
  chatDisabled: PropTypes.bool
};

export default FavoriteBadge;
