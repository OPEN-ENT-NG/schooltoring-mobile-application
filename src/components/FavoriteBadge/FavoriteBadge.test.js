import React from "react";
import { Text } from "react-native";
import { shallow } from "enzyme";

import FavoriteBadge from "./FavoriteBadge";

const onChat = jest.fn();
const onDelete = jest.fn();
const userinfo = {
  username: "Yanis ALBON",
  avatar: "/workspace/document/d4ac4110-06d5-4926-9db9-cc5ca3bcede5"
};

const rendered = shallow(
  <FavoriteBadge
    onChat={onChat}
    onDelete={onDelete}
    userinfo={userinfo}
    loading={false}
  />
);

describe("Render", () => {
  test("FavoriteBadge should render without crashing", () => {
    expect(rendered).toMatchSnapshot();
  });

  test("Rendered title should be the username in the userinfo", () => {
    expect(
      rendered
        .find(Text)
        .first()
        .contains(userinfo.username)
    ).toBeTruthy();
  });

  test(`Avatar source should be '${global.config.auth.endpoint}${
    userinfo.avatar
  }'`, () => {
    expect(
      rendered
        .find("Avatar")
        .dive()
        .find("Image")
        .props().source.uri
    ).toEqual(`${global.config.auth.endpoint}${userinfo.avatar}`);
  });
});

describe("Interactions", () => {
  test("Press onDelete button should trigger mocked function", () => {
    const onDeleteButton = rendered.find('[id="on-delete-button"]');
    onDeleteButton.simulate("press");
    expect(onDelete).toHaveBeenCalled();
  });

  test("Press onChat button should trigger mocked function", () => {
    const onChatButton = rendered.find('[id="on-chat-button"]');
    onChatButton.simulate("press");
    expect(onChat).toHaveBeenCalled();
  });
});
