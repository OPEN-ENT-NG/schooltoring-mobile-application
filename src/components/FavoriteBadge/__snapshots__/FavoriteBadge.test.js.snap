// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render FavoriteBadge should render without crashing 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "padding": 10,
      "width": "50%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "aspectRatio": 1,
        "backgroundColor": "#e3f2fd",
        "justifyContent": "center",
        "shadowColor": "#929292",
        "shadowOffset": Object {
          "height": 2,
          "width": 2,
        },
        "shadowOpacity": 0.6,
        "shadowRadius": 1,
      }
    }
  >
    <Avatar
      size={140}
      src="http://ent3-core.gdapublic.fr/workspace/document/d4ac4110-06d5-4926-9db9-cc5ca3bcede5"
    />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#fafafa",
        "maxHeight": 120,
        "shadowColor": "#929292",
        "shadowOffset": Object {
          "height": 2,
          "width": 2,
        },
        "shadowOpacity": 0.6,
        "shadowRadius": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-around",
          "padding": 5,
        }
      }
    >
      <SecondaryButton
        id="on-delete-button"
        onPress={[MockFunction]}
      >
        <Icon
          allowFontScaling={false}
          color="red"
          name="favorite"
          size={32}
        />
      </SecondaryButton>
      <SecondaryButton
        disabled={false}
        id="on-chat-button"
        onPress={[MockFunction]}
        style={
          Object {
            "backgroundColor": "#17b6b0",
            "borderRadius": 20,
            "height": 40,
            "width": 40,
          }
        }
      >
        <Icon
          allowFontScaling={false}
          color="white"
          name="chat"
          size={20}
        />
      </SecondaryButton>
    </View>
    <View
      style={
        Object {
          "paddingBottom": 15,
          "paddingHorizontal": 10,
        }
      }
    >
      <Text
        id="username-field"
        style={
          Object {
            "color": "#232323",
            "fontSize": 16,
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        Yanis ALBON
      </Text>
    </View>
  </View>
</View>
`;
